candle-transformers/src/models/depth_anything_v2.rs

```diff
@@ -0,0 +1,535 @@
+    conv1: Conv2d,
+    conv2: Conv2d,
+    batch_norm1: Option<BatchNorm>,
+    batch_norm2: Option<BatchNorm>,
+}
+
+impl ResidualConvUnit {
+    pub fn new(
+        num_features: usize,
+        activation: Activation,
+        use_batch_norm: bool,
+        var_builder: VarBuilder,
+    ) -> Result<Self> {
+        const KERNEL_SIZE: usize = 3;
+        let conv_cfg = Conv2dConfig {
+            padding: 1,
+            stride: 1,
+            dilation: 1,
+            groups: 1,
+        };
+        let conv1 = conv2d(
+            num_features,
+            num_features,
+            KERNEL_SIZE,
+            conv_cfg,
+            var_builder.push_prefix("conv1"),
+        )?;
+        let conv2 = conv2d(
+            num_features,
+            num_features,
+            KERNEL_SIZE,
+            conv_cfg,
+            var_builder.push_prefix("conv2"),
+        )?;
+
+        let (batch_norm1, batch_norm2) = match use_batch_norm {
+            true => {
+                let batch_norm_cfg = BatchNormConfig {
+                    eps: 1e-05,
+                    remove_mean: false,
+                    affine: true,
+                    momentum: 0.1,
+                };
+                (
+                    Some(batch_norm(
+                        num_features,
+                        batch_norm_cfg,
+                        var_builder.push_prefix("bn1"),
+                    )?),
+                    Some(batch_norm(
+                        num_features,
+                        batch_norm_cfg,
+                        var_builder.push_prefix("bn2"),
+                    )?),
+                )
+            }
+            false => (None, None),
+        };
+
+        Ok(Self {
+            activation,
+            conv1,
+            conv2,
+            batch_norm1,
+            batch_norm2,
+        })
+    }
+}
+
+impl Module for ResidualConvUnit {
+    fn forward(&self, xs: &Tensor) -> Result<Tensor> {
+        let out = self.activation.forward(xs)?;
+        let out = self.conv1.forward(&out)?;
+        let out = if let Some(batch_norm1) = &self.batch_norm1 {
+            batch_norm1.forward_train(&out)?
+        } else {
+            out
+        };
+
+        let out = self.activation.forward(&out)?;
+        let out = self.conv2.forward(&out)?;
+        let out = if let Some(batch_norm2) = &self.batch_norm2 {
+            batch_norm2.forward_train(&out)?
+        } else {
+            out
+        };
+
+        out + xs
+    }
+}
+
+pub struct FeatureFusionBlock {
+    res_conv_unit1: ResidualConvUnit,
+    res_conv_unit2: ResidualConvUnit,
+    output_conv: Conv2d,
+    target_patch_size: usize,
+}
+
+impl FeatureFusionBlock {
+    pub fn new(
+        num_features: usize,
+        activation: Activation,
+        use_batch_norm: bool,
+        target_patch_size: usize,
+        var_builder: VarBuilder,
+    ) -> Result<Self> {
+        const KERNEL_SIZE: usize = 1;
+        let conv_cfg = Conv2dConfig {
```

Comment:
You can use `Default::default` for the default convolution parameters rather than repeating them.

Reply:
Thanks for the tip, but padding is 1 here, not 0 like in the default.

Reply:
> Thanks for the tip, but padding is 1 here, not 0 like in the default.

```rs
        let conv_cfg = Conv2dConfig {
            padding: 1,
            ..Default::default()
        };
```

Reply:
Awesome, thanks. Will put this in the second PR